cfme/common/__init__.py

```diff
@@ -79,14 +84,51 @@ def add_tag(self, tag, single_value=False):
         self.load_details(refresh=True)
         mixins.add_tag(tag, single_value=single_value, navigate=True)
 
+    def add_tags(self, tags):
+        """Add list of tags
+        tags: List of `Tag` namedtuple
+        """
+        for _tag in tags:
+            self.add_tag(tag=[_tag.category, _tag.tag_name], single_value=_tag.single_value)
+
     def remove_tag(self, tag):
         self.load_details(refresh=True)
         mixins.remove_tag(tag)
 
+    def remove_tags(self, tags):
+        """Remove list of tags
+        tags: List of `Tag` namedtuple
+        """
+        for _tag in tags:
+            self.remove_tag(tag=[_tag.category, _tag.tag_name])
+
+    @variable(alias='ui')
     def get_tags(self, tag="My Company Tags"):
         self.load_details(refresh=True)
         return mixins.get_tags(tag=tag)
 
+    @get_tags.variant('db')
+    def get_tags_db(self):
+        """
+        Gets tags detail from database
+        Column order: `tag_id`, `db_id`, `category`, `tag_name`, `single_value`
+        """
+        if not self.db_id:
+            self.load_details()
+        if not self.db_id or not self.taggable_type:
+            raise KeyError("'db_id' or 'taggable_type' not set")
+        t_cls1 = aliased(cfmedb()['classifications'])
+        t_cls2 = aliased(cfmedb()['classifications'])
+        t_tgg = aliased(cfmedb()['taggings'])
+        query = cfmedb().session.query(t_cls1.tag_id, t_tgg.taggable_id.label('db_id'),
+                                       t_cls2.description.label('category'),
+                                       t_cls1.description.label('tag_name'), t_cls1.single_value)\
+            .join(t_cls2, t_cls1.parent_id == t_cls2.id)\
+            .join(t_tgg, t_tgg.tag_id == t_cls1.tag_id)\
+            .filter(t_tgg.taggable_id == self.db_id)\
+            .filter(t_tgg.taggable_type == self.taggable_type)
+        return query.all()
```

Comment:
Right - I'm unsure if the way forward will change - ultimately we will want it to return "Tag" objects, so that they are usable elsewhere. We have not done this in the past, and it will be something new, I would say, if you can lets make it collapse to the GUI format for compatibility purposes, but that in the future we are already thinking about how to return objects, possibly by querying the db to help instantiate with the correct information.